src/formatDate.js

```diff
@@ -8,7 +8,33 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+  const deleteLastElement = fromFormatCopy.pop();
+  const partOfDate = date.split(deleteLastElement);
+  const partsOfDate = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    partsOfDate[part] = partOfDate[index];
+  });
+
+  if (toFormatCopy.includes('YY') && partsOfDate['YYYY']) {
+    partsOfDate['YY'] = partsOfDate['YYYY'].slice(-2);
+  }
+
+  if (toFormatCopy.includes('YYYY') && partsOfDate['YY']) {
+    partsOfDate['YYYY'] =
+      partsOfDate['YY'] < '30'
+        ? '20' + partsOfDate['YY']
+        : '19' + partsOfDate['YY'];
+  }
```

Comment:
The conversion from 'YY' to 'YYYY' is based on a hardcoded threshold ('30'), which might not be suitable for all use cases. Consider making this logic more flexible or configurable.